### PR TITLE
Improve deletion error message

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

This change trims the deletions message for large deletion failures

### Why are the changes needed?

When many inodes fail to be deleted, the error msg will be too long for the log message and generate too many side product strings.

### Does this PR introduce any user facing changes?

The error msg users see will be different. Although the message itself is trimmed, I don't think the user will need the full list of all failure messages in one go. My estimation is the first batch of error messages will be representative enough and if not, after fixing the first batch, the next try will reveal the left batch of failures and eventually all failures can be fixed.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -2010,11 +2010,21 @@ public void deleteInternal(RpcContext rpcContext, LockedInodePath inodePath,
       }
 
       if (!failedUris.isEmpty()) {
-        Collection<String> messages = failedUris.stream()
-            .map(pair -> String.format("%s (%s)", pair.getFirst(), pair.getSecond()))
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:2014]: The map-collect will generate too many small strings. Also on the collection phase, the ArrayList will have auto-sizing and many array copies. The collected list will be joined into one string at last, so the streaming is essentially just a syntactic sugar resource waste.

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 2]
```diff
@@ -2010,11 +2010,21 @@ public void deleteInternal(RpcContext rpcContext, LockedInodePath inodePath,
...
-            .collect(Collectors.toList());
-        throw new FailedPreconditionException(
-            ExceptionMessage.DELETE_FAILED_UFS.getMessage(StringUtils.join(messages, ", ")));
+        StringBuilder errorReport = new StringBuilder(
+            ExceptionMessage.DELETE_FAILED_UFS.getMessage(failedUris.size() + " paths: "));
+        boolean trim = failedUris.size() > 20;
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:2015]: If 20 is not enough, we can change it to 100. For numbers more than 100, I don't think people will read those additional information carefully.

**Review by ZhuTopher — COMMENTED** [review]
I think the changes make sense, especially the point about streaming -> map being resource intensive.

My only question is whether or not it be worth having `trim` be affected by class-level log levels. If `DEBUG` we keep all messages, otherwise we have the hardcoded limits here (20 or otherwise).

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java (hunk 2 above)
Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:2015]: Could you move this logic to a private method? `buildDeleteFailureMessage()`?

**Review by ggezer — APPROVED** [review]

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java (hunk 2 above)
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:2015]: Done

**Comment by jiacheliu3** [comment]
> I think the changes make sense, especially the point about streaming -> map being resource intensive.
> 
> My only question is whether or not it be worth having `trim` be affected by class-level log levels. If `DEBUG` we keep all messages, otherwise we have the hardcoded limits here (20 or otherwise).

Yea added. I hope the full list can provide more value to people than I anticipate.

Although a side note based on my personal experience is, every time I try to see the full content of a long error list, either:
(1) The first few reveal the problem already and prove me an idiot
(2) I still have no idea and give up at the first batch, proving me a bigger idiot

**Comment by jiacheliu3** [comment]
The failed test is a known flake:
```
Error: 8.643 [ERROR] alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats  Time elapsed: 45.324 s  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for job 1646294802233 to be one of status [CANCELED] options: WaitForOptions{interval=20, timeout=45000} last value: false
	at alluxio.util.CommonUtils.waitForResult(CommonUtils.java:384)
	at alluxio.util.CommonUtils.waitFor(CommonUtils.java:342)
	at alluxio.job.util.JobTestUtils.waitForJobStatus(JobTestUtils.java:96)
	at alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats(DistributedCommandsStatsTest.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at alluxio.testutils.LocalAlluxioClusterResource$ResetRule$1.evaluate(LocalAlluxioClusterResource.java:319)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Review by ZhuTopher — APPROVED** [review]
Thanks for the changes!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**